phpBB/includes/functions.php

```diff
@@ -2833,7 +2848,7 @@ function login_box($redirect = '', $l_explain = '', $l_success = '', $admin = fa
 		trigger_error('NO_AUTH_ADMIN');
 	}
 
-	if ($request->is_set_post('login') || ($request->is_set('login') && $request->variable('login', '') == 'external'))
+	if (($request->is_set_post('login') || ($request->is_set('login') && $request->variable('login', '') == 'external')) && !$err)
```

Comment:
Please use empty() and move the check for err being set in front of the other checks.